tests/src/test/java/com/maxio/advancedbilling/controllers/coupons/CouponsControllerListTest.java

```diff
@@ -0,0 +1,180 @@
+                                            .build()
+                            ).build()
+                )
+                .stream().map(CouponResponse::getCoupon).toList();
+
+        assertThat(listedCoupons).hasSize(10);
+        assertThat(listedCoupons).usingRecursiveFieldByFieldElementComparator()
+                .containsExactlyInAnyOrderElementsOf(coupons);
+    }
+
+    @Test
+    void shouldListCouponsWithIdFilters() throws IOException, ApiException {
+        List<Coupon> couponsToFind = coupons.subList(2,8);
+        List<Coupon> listedCoupons  = COUPONS_CONTROLLER.listCoupons(
+                        new ListCouponsInput.Builder()
+                                .filter(
+                                        new ListCouponsFilter.Builder()
+                                                .ids(couponsToFind.stream().map(Coupon::getId).toList())
+                                                .build()
+                                ).build()
+                )
+                .stream().map(CouponResponse::getCoupon).toList();
+
+        assertThat(listedCoupons).hasSize(6);
+        assertThat(listedCoupons).usingRecursiveFieldByFieldElementComparator()
+                .containsExactlyInAnyOrderElementsOf(couponsToFind);
+    }
+
+    @Test
+    void shouldListCouponsWithCodeFilters() throws IOException, ApiException {
+        List<Coupon> couponsToFind = coupons.subList(5,9);
+        List<Coupon> listedCoupons  = COUPONS_CONTROLLER.listCoupons(
+                        new ListCouponsInput.Builder()
+                                .filter(
+                                        new ListCouponsFilter.Builder()
+                                                .codes(couponsToFind.stream().map(Coupon::getCode).toList())
+                                                .build()
+                                ).build()
+                )
+                .stream().map(CouponResponse::getCoupon).toList();
+
+        assertThat(listedCoupons).hasSize(4);
+        assertThat(listedCoupons).usingRecursiveFieldByFieldElementComparator()
+                .containsExactlyInAnyOrderElementsOf(couponsToFind);
+    }
+
+    @Test
+    void shouldListAllCouponsWithPaging() throws IOException, ApiException {
+        // when
+        List<Coupon> allResponses = new ArrayList<>();
+        List<Coupon> firstPage = listCouponsWithPaging(1, 4);
+        List<Coupon> secondPage = listCouponsWithPaging(2, 4);
+        List<Coupon> thirdPage = listCouponsWithPaging(3, 4);
+        List<Coupon> fourthPage = listCouponsWithPaging(4, 4);
+
+        allResponses.addAll(firstPage);
+        allResponses.addAll(secondPage);
+        allResponses.addAll(thirdPage);
+
+
+        // then
+        assertThat(firstPage).hasSize(4);
+        assertThat(secondPage).hasSize(4);
+        assertThat(thirdPage).hasSize(2);
+        assertThat(fourthPage).isEmpty();
+
+        assertThat(allResponses)
```

Comment:
Isn't it possible to get in response coupons created in other test? And then this assertion will fail?